pkg/updater/download.go

```diff
@@ -42,149 +45,111 @@ func newDownloader(client *http.Client) *downloader {
 	}
 }
 
-// Download downloads the package at the given URL in temporary directory,
-// verifies its SHA256 hash and extracts it to the given destination path.
-// It currently assumes the package is a tar.gz archive.
-func (d *downloader) Download(ctx context.Context, pkg Package, destinationPath string) error {
+// Download downloads the Datadog Package referenced in the given Package struct.
+func (d *downloader) Download(ctx context.Context, tmpDir string, pkg Package) (oci.Image, error) {
 	log.Debugf("Downloading package %s version %s from %s", pkg.Name, pkg.Version, pkg.URL)
 
-	// Create temporary directory to download the archive
-	tmpDir, err := os.MkdirTemp("", "")
+	// TODO: Add support for OCI registries
+	image, err := d.downloadFromURL(ctx, pkg.URL, pkg.SHA256, pkg.Size, tmpDir)
 	if err != nil {
-		return fmt.Errorf("could not create temporary directory: %w", err)
+		return nil, fmt.Errorf("could not download package from %s: %w", pkg.URL, err)
 	}
-	defer func() {
-		err := os.RemoveAll(tmpDir)
-		if err != nil {
-			log.Errorf("could not cleanup temporary directory: %v", err)
-		}
-	}()
 
-	// Download archive
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, pkg.URL, nil)
+	log.Debugf("Successfully downloaded package %s version %s from %s", pkg.Name, pkg.Version, pkg.URL)
+	return image, nil
+}
+
+func (d *downloader) downloadFromURL(ctx context.Context, url string, sha256hash string, size int64, tmpDir string) (oci.Image, error) {
+	// Request the oci-layout.tar archive from the given URL
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
 	if err != nil {
-		return fmt.Errorf("could not create download request: %w", err)
+		return nil, fmt.Errorf("could not create download request: %w", err)
 	}
 	resp, err := d.client.Do(req)
 	if err != nil {
-		return fmt.Errorf("could not download package: %w", err)
+		return nil, fmt.Errorf("could not download package: %w", err)
 	}
 	defer resp.Body.Close()
-	if resp.ContentLength != pkg.Size {
-		return fmt.Errorf("invalid size for %s: expected %d, got %d", pkg.URL, pkg.Size, resp.ContentLength)
+	if resp.ContentLength != -1 && resp.ContentLength != size {
+		return nil, fmt.Errorf("invalid size: expected %d, got %d", size, resp.ContentLength)
 	}
 
-	// Write archive on disk & check its hash while doing so
+	// Write oci-layout.tar on disk and check its hash while doing so
 	hashWriter := sha256.New()
 	reader := io.TeeReader(
-		io.LimitReader(resp.Body, maxArchiveSize),
+		io.LimitReader(resp.Body, ociLayoutArchiveMaxSize),
 		hashWriter,
 	)
-	archivePath := filepath.Join(tmpDir, archiveName)
-
-	// Check there is enough space to write the compressed archive
-	enoughSpace, err := tools.CheckAvailableDiskSpace(filepath.Dir(archivePath), uint64(pkg.Size))
+	ociLayoutArchivePath := filepath.Join(tmpDir, ociLayoutArchiveName)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏Only used once, we could directly set it in the create